File: 49-group-anagrams/group-anagrams.go
func SortString(w string) string {
    s := strings.Split(w, "")
    sort.Strings(s)
    return strings.Join(s, "")
}
func groupAnagrams(strs []string) [][]string {
    result := [][]string{}
    anaMap := map[string][]int{}
    for index, s := range strs {
        key := SortString(s)
        anaMap[key] = append(anaMap[key], index)
    }
    for _, vals := range anaMap {
        temp := []string{}
        for _,index := range vals {
            temp = append(temp, strs[index])
        }
        result = append(result, temp)
    }
    return result
}

File: 49-group-anagrams/group-anagrams.py
class Solution:
    def groupAnagrams(self, strs: List[str]) -> List[List[str]]:
        # Strategy is to sort each string and then check for anagram
        # Next is to form groups with original strings by index based storage
        result = []
        anaMap = {}
        for index, s in enumerate(strs):
            subStr = ''.join(sorted(s))
            if subStr in anaMap:
                anaMap[subStr].append(index)
            else:
                anaMap[subStr] = [index]
        for val in anaMap.values():
            temp = []
            print(len(s))
            for index in val:
                temp.append(strs[index])
            result.append(temp)
        return result

        